src/main/java/teammates/storage/sqlentity/FeedbackSession.java

```diff
@@ -112,6 +112,35 @@ public FeedbackSession(String name, Course course, String creatorEmail, String i
         this.setPublishedEmailEnabled(isPublishedEmailEnabled);
     }
 
+    /**
+     * Creates a copy that uses the specific deadline for the given user.
+     *
+     * @param userEmail The email address of the given user.
+     * @return The copy of this object for the given user.
+     */
+    public FeedbackSession getCopyForUser(String userEmail) {
+        FeedbackSession copy = getCopy();
+        for (DeadlineExtension de : copy.getDeadlineExtensions()) {
+            if (!de.getUser().getEmail().equals(userEmail)) {
+                de.setEndTime(copy.getEndTime());
+            }
+        }
+        return copy;
+    }
+
+    private FeedbackSession getCopy() {
+        FeedbackSession fs = new FeedbackSession(
+                name, course, creatorEmail, instructions, startTime,
+                endTime, sessionVisibleFromTime, resultsVisibleFromTime,
+                gracePeriod, isOpeningEmailEnabled, isClosingEmailEnabled, isPublishedEmailEnabled
+        );
+
+        fs.setCreatedAt(getCreatedAt());
+        fs.setDeletedAt(getDeletedAt());
+
```

Comment:
Setting the deadlineExtension here should solve the problem. Also set the `updatedAt` just to be safer.

Reply:
alright, got it! i've set `updatedAt` and deadlineExtensions as well here